perception/autoware_traffic_light_classifier/src/traffic_light_classifier_node.cpp

```diff
@@ -95,41 +98,49 @@ void TrafficLightClassifierNodelet::imageRoiCallback(
   }
 
   tier4_perception_msgs::msg::TrafficLightArray output_msg;
-
   output_msg.signals.resize(input_rois_msg->rois.size());
 
   std::vector<cv::Mat> images;
   std::vector<size_t> backlight_indices;
   for (size_t i = 0; i < input_rois_msg->rois.size(); i++) {
-    // skip if the roi is not detected
-    if (input_rois_msg->rois.at(i).roi.height == 0) {
-      break;
+    const auto & input_roi = input_rois_msg->rois.at(i);
+
+    // ignore if the roi is not the type to be classified
+    if (input_roi.traffic_light_type != classify_traffic_light_type_) {
+      continue;
     }
-    if (input_rois_msg->rois.at(i).traffic_light_type != classify_traffic_light_type_) {
+    // skip if the roi size is zero
+    if (input_roi.roi.height == 0 || input_roi.roi.width == 0) {
       continue;
     }
-    output_msg.signals[images.size()].traffic_light_id =
-      input_rois_msg->rois.at(i).traffic_light_id;
-    output_msg.signals[images.size()].traffic_light_type =
-      input_rois_msg->rois.at(i).traffic_light_type;
-    const sensor_msgs::msg::RegionOfInterest & roi = input_rois_msg->rois.at(i).roi;
 
+    // set traffic light id and type
+    output_msg.signals[images.size()].traffic_light_id = input_roi.traffic_light_id;
+    output_msg.signals[images.size()].traffic_light_type = input_roi.traffic_light_type;
```

Comment:
This might not be directly related to the purpose of the current changes, so it's just a comment. I believe the reason for using images.size() as the index is that it returns the correct index, given that images.emplace_back(roi_img) is called afterward. However, I think this code is a bit tricky.

Reply:
I feel the same, and tried to fix it. However, that caused error and I returned back to the original implementation.

Reply:
fixed in https://github.com/autowarefoundation/autoware.universe/pull/7608/commits/464d1f88945a3384427ff45bc11f5d575b9b9d41